### PR TITLE
Added an in-game encyclopedia to Dune 2000

### LINKED ISSUE #3788: Encyclopedia
Something that might take a while to add but would be totaly worth it imo.

In Dune 2 you could check an encyclopedia with all units in that game.

We could use something like that in openRA (it wont be much use when you play multiplayer, but it will be a nice touch imo)

A good example here is the game "Battle for Wesnoth".

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/3788.

### CODE REVIEW

**Comment by pchote** [comment]
I really don't like the idea of hardcoding encyclopedia hooks into individual traits. This makes it really inflexible for modders, forcing everything to follow a single standard pattern for descriptions.

**Comment by Mailaender** [comment]
I started out with something more obvious:

```yaml
Encyclopedia:
    Armor:
    Weapon:
    Speed:
    Range:
    Purpose:
```

but that turned out to not really match the self describing OpenRA data structure and be too inflexible for special units.

**Comment by pchote** [comment]
The end result of the implementation is displaying paragraphs of free-form text. Nothing here seems to need to know which sentence describes the armor, which describes the speed, and so on. This could be done much more cleanly with a single EncyclopediaData trait that has a single Text field (which would eventually just be a reference to an entry in the ftl files)

**Comment by Mailaender** [comment]
That is how Dune 2000 stored everything. My experience with no or very loose data structures is that it will only cause junk data in the end. You are indeed right that it probably simplifies things a lot and the end result is the same. Updated.

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs [hunk 1]
```diff
@@ -11,7 +11,6 @@
 
 using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Could we maybe split the unrelated cleanup to another PR? (although I imagine it's likely much of it would disappear with a rebase anyway...)

File: OpenRA.Mods.Common/Traits/Encyclopedia.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class EncyclopediaInfo : TraitInfo
+	{
+		[Desc("Explains the purpose in the in-game encyclopedia.")]
+		public readonly string Description = null;
+
+		public override object Create(ActorInitializer init) { return new Encyclopedia(); }
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Traits/Encyclopedia.cs]: Since `Encyclopedia` has no behaviour, could we make it a singleton to avoid allocations?
e.g.
```c#
public class Encyclopedia
{
    public static readonly Encyclopedia Instance = new Encyclopedia();
    private Encyclopedia() {}
}
```

File: OpenRA.Mods.Common/Widgets/Logic/EncyclopediaLogic.cs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+			widget.Get<ButtonWidget>("BACK_BUTTON").OnClick = () =>
+			{
+				Game.Disconnect();
+				Ui.CloseWindow();
+				onExit();
+			};
+		}
+
+		bool disposed;
+		protected override void Dispose(bool disposing)
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/Widgets/Logic/EncyclopediaLogic.cs]: No need to override `Dispose` or have `bool disposed` if there's nothing to do.

File: mods/d2k/rules/aircraft.yaml [hunk 4]
```diff
@@ -95,13 +98,17 @@ frigate:
...
+		Prerequisites: palace
 	AttackBomber:
 	Armament:
 		Weapon: OrniBomb
 	Health:
 		HP: 9000
 	Armor:
 		Type: light
+	Encyclopedia:
+		Description: The fastest aircraft on Dune, the Ornithopther is lightly armored and capable of dropping 500lb bombs.  This unit is most effective against infantry and lightly armored targets, but also damages armored targets.\n\nThey are vulnerable to missiles and can only be hit by anti-aircraft weapon.
```
Comment (RoosterDragon) [inline mods/d2k/rules/aircraft.yaml]: Here (after `bombs.`) and throughout there are double spaces after the end of some sentences, should those be single spaced instead?

File: mods/d2k/rules/infantry.yaml [hunk 5]
```diff
@@ -48,7 +50,9 @@ engineer:
 	CaptureManager:
 	Captures:
 		CaptureTypes: building
-		PlayerExperience: 50
+		PlayerExperience: 50#
```
Comment (RoosterDragon) [inline mods/d2k/rules/infantry.yaml]: Stray `#` here.

File: mods/d2k/rules/infantry.yaml [hunk 6]
```diff
@@ -48,7 +50,9 @@ engineer:
...
+	Encyclopedia:
+		Description: Engineers can be used to capture enemy buildings.\n\n Engineers are resistant to anti-tank weaponry but very vulnerable to high explosives, fire and bullet weapons.
```
Comment (RoosterDragon) [inline mods/d2k/rules/infantry.yaml]: Nit: Space after `\n\n `

File: mods/d2k/rules/infantry.yaml [hunk 7]
```diff
@@ -160,6 +168,8 @@ fremen:
 		Weapon: Fremen_S
 	Armament@SECONDARY:
 		Weapon: Fremen_L
+	Encyclopedia:
+		Description:  Fremen are the native desert warriors of Dune.  Fremen ground units carry 10mm Assault Rifles and Rockets.  Their firepower is equally effective against infantry and vehicles.\n\nFremen units are very vulnerable to high explosive and bullet weapons.
```
Comment (RoosterDragon) [inline mods/d2k/rules/infantry.yaml]: Nit: Double space after `Description:`

File: mods/d2k/rules/structures.yaml [hunk 8]
```diff
@@ -951,6 +981,8 @@ research_centre:
 		Footprint: _X_ xxx XXX ===
 		Dimensions: 3,4
 		LocalCenterOffset: 0,-512,0
+	Encyclopedia:
+		Description: The Ix Research Center provides technology upgrades for structures and vehicles.  This facility is required for production of a number of advanced special weapons and prototypes.\n\nThe Ix Research Center is vulnerable to most types of weapons.
```
Comment (RoosterDragon) [inline mods/d2k/rules/structures.yaml]: Should it be capitalised `IX` like the tooltip?

File: mods/d2k/rules/vehicles.yaml [hunk 9]
```diff
@@ -407,6 +423,8 @@ raider:
 		Cost: 350
 	Tooltip:
 		Name: Raider Trike
+	Encyclopedia:
+		Description: Raiders are similar to Trikes, but the Ordos have refined their fire power, speed and armor to create a powerful and maneuverable  scout.  With dual 20mm cannons, Raiders are most effective against infantry and lightly armored vehicles.\n\nRaiders are vulnerable to most types of weaponry, though high caliber guns are slightly less effective.
```
Comment (RoosterDragon) [inline mods/d2k/rules/vehicles.yaml]: Nit: Double space after `maneuverable`.

File: mods/d2k/rules/structures.yaml [hunk 10]
```diff
@@ -770,6 +794,8 @@ large_gun_turret:
 		Cost: 750
 	Tooltip:
 		Name: Rocket Turret
+	Encyclopedia:
+		Description: Purpose: Base defense. The substantially improved Rocket Turret has a longer range and a higher rate of fire than the Gun Turret. The Rocket Turret's advanced targeting equipment requires power to operate.\n\n  The Rocket Turret is resistant to bullet and explosive weapons, but vulnerable to missiles and high caliber guns.
```
Comment (RoosterDragon) [inline mods/d2k/rules/structures.yaml]: Nit: Extra space after `\n\n`

File: mods/d2k/rules/vehicles.yaml [hunk 11]
```diff
@@ -175,6 +181,8 @@ quad:
 	Armament:
 		Weapon: Rocket
 		LocalOffset: 128,64,64, 128,-64,64
+	Encyclopedia:
+		Description: Stronger than the Trike in both armor and firepower, the Quad is a four-wheeled vehicle firing armor-piercing rockets.  The Quad is effective against most vehicles.\n\n Quads are resistant to bullets and high explosives, to a lesser degree.  However, Quads are vulnerable to missiles and high caliber guns.
```
Comment (RoosterDragon) [inline mods/d2k/rules/vehicles.yaml]: Nit: Extra space after `\n\n`

**Review by RoosterDragon — CHANGES_REQUESTED** [review]

**Comment by Mailaender** [comment]
Rebased and updated.

File: mods/d2k/rules/infantry.yaml [hunk 12]
```diff
@@ -19,6 +19,8 @@ light_inf:
 		Speed: 43
 	Armament:
 		Weapon: LMG
+	Encyclopedia:
+		Description: Light Infantry are lightly armored foot soldiers, equipped with 9mm RP assault rifles.  Light Infantry are effective against other infantry and lightly armored vehicles.\n\nLight Infantry are resistant to missiles and large caliber guns, but very vulnerable to high explosives, fire and bullet weapons.
```
Comment (RoosterDragon) [inline mods/d2k/rules/infantry.yaml]: What was your preference between single or double spacing between sentences? There's a mix throughout so whatever we go for will need amending across the files.

File: mods/d2k/rules/structures.yaml [hunk 13]
```diff
@@ -953,6 +983,8 @@ research_centre:
 		Footprint: _X_ xxx XXX ===
 		Dimensions: 3,4
 		LocalCenterOffset: 0,-512,0
+	Encyclopedia:
+		Description: The IX Research Center provides technology upgrades for structures and vehicles.  This facility is required for production of a number of advanced special weapons and prototypes.\n\nThe Ix Research Center is vulnerable to most types of weapons.
```
Comment (RoosterDragon) [inline mods/d2k/rules/structures.yaml]: There's a second `Ix Research Center` in this desc that needs the capitalisation fix.

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs (hunk 1 above)
Comment (Mailaender) [reply OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: There are still many issues left. I added them to https://github.com/OpenRA/OpenRA/pull/19953.

File: mods/d2k/rules/infantry.yaml (hunk 12 above)
Comment (Mailaender) [reply mods/d2k/rules/infantry.yaml]: I don't remember adding them on purpose, so I removed them.

File: mods/d2k/rules/structures.yaml (hunk 13 above)
Comment (Mailaender) [reply mods/d2k/rules/structures.yaml]: Done.

**Review by RoosterDragon — DISMISSED** [review]

File: mods/d2k/rules/structures.yaml [hunk 14]
```diff
@@ -735,6 +757,8 @@ medium_gun_turret:
 		DecorationBounds: 1024, 1280, 0, -256
 	Health:
 		HP: 27000
+	Encyclopedia:
+		Description: Purpose: Base defense. The Gun Turret has a medium range gun which is effective against vehicles, especially heavily armored vehicles. The Gun Turret will fire on any enemy unit within range.\n\nThe Gun Turret is resistant to bullet and explosive weapons, but vulnerable to missiles and high-caliber guns.
```
Comment (MustaphaTR) [inline mods/d2k/rules/structures.yaml]: This and Rocket Turret has `Purpose:` that you seem to have removed from the other descriptions.

File: mods/d2k/rules/aircraft.yaml [hunk 15]
```diff
@@ -103,6 +108,8 @@ ornithopter:
 		HP: 9000
 	Armor:
 		Type: light
+	Encyclopedia:
+		Description: The fastest aircraft on Dune, the Ornithopther is lightly armored and capable of dropping 500lb bombs. This unit is most effective against infantry and lightly armored targets, but also damages armored targets.\n\nThey are vulnerable to missiles and can only be hit by anti-aircraft weapon.
```
Comment (MustaphaTR) [inline mods/d2k/rules/aircraft.yaml]: Ornithopters are complately untargetable on OpenRA.

File: mods/d2k/rules/infantry.yaml [hunk 16]
```diff
@@ -276,6 +292,8 @@ saboteur:
 		AddToArmyValue: true
 	Health:
 		HP: 4000
+	Encyclopedia:
+		Description: The Saboteur is resistant to anti-tank weaponry, but very vulnerable to high explosives, fire, and bullet weapons.\n\nThe Saboteur is a special military unit acquired by House Ordos. A single Saboteur can destroy any enemy building once he moves into it, though also destroys himself! A Saboteur can be stealthed by deploying itself.
```
Comment (MustaphaTR) [inline mods/d2k/rules/infantry.yaml]: The order of main info and armor info is switched here.

File: mods/d2k/rules/structures.yaml [hunk 17]
```diff
@@ -125,6 +129,8 @@ wind_trap:
 		Footprint: xx xx ==
 		Dimensions: 2,3
 		LocalCenterOffset: 0,-512,0
+	Encyclopedia:
+		Description: Wind Traps provide power and water to an installation. Large, above-ground ducts funnel wind currents underground into massive turbines which power generators and humidity extractors.\n\nWind Traps are vulnerable to most types of weapons.
```
Comment (MustaphaTR) [inline mods/d2k/rules/structures.yaml:139]: Description for Wind Trap says it requires Construction Yard in the original game. Not sure if we should add it here or not tho. ConYard's desc also says it doesn't require anything but is built by deploying an MCV.

Similar thing is the case with Light Inf and Barracks, Trike/Railder with Light Factory and Combat Tank with Heavy Factory.

File: mods/d2k/rules/structures.yaml [hunk 18]
```diff
@@ -5,6 +5,8 @@
 		TerrainTypes: Rock
 		BuildSounds: CHUNG.WAV
 		AllowInvalidPlacement: true
+	Encyclopedia:
+		Description: If buildings are not placed on concrete, they will be damaged. Buildings can be repaired, but unless the building sits completely on concrete, the building will suffer continual weathering damage from the erosive desert environment.\n\nConcrete is vulnerable to most weapon types. Concrete cannot be repaired if damaged.
```
Comment (MustaphaTR) [inline mods/d2k/rules/structures.yaml]: Concrete doesn't seem to be appearing on the list.

File: mods/d2k/rules/infantry.yaml [hunk 19]
```diff
@@ -260,6 +272,10 @@ mpsardaukar:
 		Weapon: M_HMG_H
 	RenderSprites:
 		Image: sardaukar
+	UpdatesPlayerStatistics:
+		OverrideActor: sardaukar
+	Encyclopedia:
+		Description: These powerful heavy troopers have a machine gun that's effective against infantry, and a rocket launcher for vehicles.
```
Comment (MustaphaTR) [inline mods/d2k/rules/infantry.yaml]: Sardaukar doesn't seem to appearing in the list either.

File: OpenRA.Mods.Common/Traits/Encyclopedia.cs [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class EncyclopediaInfo : TraitInfo
+	{
+		[Desc("Explains the purpose in the in-game encyclopedia.")]
+		public readonly string Description = null;
+
```
Comment (MustaphaTR) [inline OpenRA.Mods.Common/Traits/Encyclopedia.cs:20]: The order of the Units section is a mess. Would be nice if we had something in the Encyclopedia trait to order them as we want.

I think the order should be as following:
Light Inf, Trooper, Engineer, Thumper, Grenadier, Sardaukar, Fremen, Saboteur, Trike, Raider, Quad, Stealth Raider, Harvester, Atr. Combat Tank, Ord. Combat Tank, Har. Combat Tank, Siege Tank, MCV, Missile Tank, Sonic Tank, Deviator, Devastator, Carryall, Ornithopter, Death Hand.

File: OpenRA.Mods.Common/Widgets/Logic/EncyclopediaLogic.cs [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+					continue;
+
+				if (actor.TraitInfoOrDefault<BuildingInfo>() != null)
+					buildings.Add(actor);
+				else
+					units.Add(actor);
+			}
+
+			CreateActorGroup("Buildings", buildings);
+			CreateActorGroup("Units", units);
```
Comment (MustaphaTR) [inline OpenRA.Mods.Common/Widgets/Logic/EncyclopediaLogic.cs]: I think it would be better if these weren't hardcoded but defined in Encyclopedia trait. Another mod that may want to use this logic may want to split Units to Infantry/Vehicles/Aircraft etc or Buildings to Buildings/Defences.

File: mods/d2k/rules/vehicles.yaml [hunk 22]
```diff
@@ -265,6 +275,8 @@ missile_tank:
 		HP: 13000
 	Armor:
 		Type: wood
+	Encyclopedia:
+		Description: It is anti-aircraft capable and effective against most targets, except infantry units.\n\nMissile Tanks are vulnerable to most weapons, though high caliber guns are slightly less effective.
```
Comment (MustaphaTR) [inline mods/d2k/rules/vehicles.yaml]: I know we are mostly copying from the original descs, but using "it" here feels weird, most other descs start with the unit's name. So `Missile Tanks are` or `The Missile Tank is`.

File: mods/d2k/rules/vehicles.yaml [hunk 23]
```diff
@@ -415,6 +431,8 @@ raider:
 		Cost: 350
 	Tooltip:
 		Name: Raider Trike
+	Encyclopedia:
+		Description: Raiders are similar to Trikes, but the Ordos have refined their fire power, speed and armor to create a powerful and maneuverable scout. With dual 20mm cannons, Raiders are most effective against infantry and lightly armored vehicles.\n\nRaiders are vulnerable to most types of weaponry, though high caliber guns are slightly less effective.
```
Comment (MustaphaTR) [inline mods/d2k/rules/vehicles.yaml:451]: Stealth Raider is inheriting this and appearing with the same desc, apart from the prerequisites. It needs its own one.

File: mods/d2k/rules/structures.yaml (hunk 17 above)
Comment (Mailaender) [reply mods/d2k/rules/structures.yaml:139]: It seemed redundant. Without a construction yard, you can't build anything anyway.

**Comment by Mailaender** [comment]
Updated with all the suggested flexibility.

File: mods/d2k/rules/aircraft.yaml [hunk 24]
```diff
@@ -103,6 +109,9 @@ ornithopter:
 		HP: 9000
 	Armor:
 		Type: light
+	Encyclopedia:
+		Description: The fastest aircraft on Dune, the Ornithopther is lightly armored and capable of dropping 500lb bombs. This unit is most effective against infantry and lightly armored targets, but also damages armored targets.\n\nThey fly fast and are not targetable by anti-aircraft weapon.
+		Order: 240
```
Comment (MustaphaTR) [inline mods/d2k/rules/aircraft.yaml:114]: You seem to have forgotten to add Category definition to the Aircraft.

File: mods/d2k/rules/aircraft.yaml [hunk 25]
```diff
@@ -95,6 +99,8 @@ frigate:
 
 ornithopter:
 	Inherits: ^Plane
+	Buildable:
+		Prerequisites: palace
```
Comment (MustaphaTR) [inline mods/d2k/rules/aircraft.yaml]: Ornithopter ability requires High Tech Factory Upgrade, not Palace.

File: mods/d2k/rules/aircraft.yaml [hunk 26]
```diff
@@ -70,6 +73,7 @@ carryall:
 	Buildable:
 		Queue: Aircraft
 		BuildPaletteOrder: 120
+		Prerequisites: high_tech_factory
```
Comment (MustaphaTR) [inline mods/d2k/rules/aircraft.yaml]: If other units (Wind Trap, Concrete, Trike, Raider, Combat Tank, Light Inf.) that are directly available from their production structures don't have that requirement shown, i don't think Carryall should either.

File: mods/d2k/rules/misc.yaml [hunk 27]
```diff
@@ -331,3 +331,17 @@ upgrade.hightech:
...
 	ProvidesPrerequisite@upgradename:
+
+deathhand:
+	AlwaysVisible:
+	Interactable:
+	Tooltip:
+		Name: Death Hand
+	Encyclopedia:
+		Description: The Death Hand Missiles' warhead carries atomic cluster munitions. It detonates above the target, inflicting great damage over a wide area.
+		Order: 250
```
Comment (MustaphaTR) [inline mods/d2k/rules/misc.yaml:343]: This is also missing Category.

File: mods/d2k/rules/structures.yaml [hunk 28]
```diff
@@ -61,6 +66,9 @@ construction_yard:
 		Dimensions: 3,3
 		LocalCenterOffset: 0,-512,0
 		-ConcretePrerequisites:
+	Encyclopedia:
+		Description: The Construction Yard is the foundation of any base built on Arrakis. Construction Yards produce a small amount of power and are required for the building of any new structures. Protect this structure! It is critical to the success of your base.\n\nConstruction yards are fairly strong, but vulnerable in varying degrees to all types of weapons.
+		Category: Buildings
```
Comment (MustaphaTR) [inline mods/d2k/rules/structures.yaml:72]: ConYard should appear before Concrete on the Encyclopedia imo, so i guess Buildings will need Order too.

File: mods/d2k/rules/misc.yaml (hunk 27 above)
Comment (Mailaender) [reply mods/d2k/rules/misc.yaml:343]: This is on purpose.

**Comment by Mailaender** [comment]
Updated.

File: mods/d2k/rules/aircraft.yaml (hunk 24 above)
Comment (Mailaender) [reply mods/d2k/rules/aircraft.yaml:114]: This is actually somewhat in between unit and super power.

File: mods/d2k/rules/misc.yaml (hunk 27 above)
Comment (MustaphaTR) [reply mods/d2k/rules/misc.yaml:343]: Nameless categoty looks weird, if you want a seperate category, perhaps put This and Orthithopter under a Support Powers category.

Comment (Mailaender) [reply mods/d2k/rules/misc.yaml:343]: That requires sorting of the categories, which I now added as well.

**Review by RoosterDragon — DISMISSED** [review]

**Comment by penev92** [comment]
I see a bunch of seemingly random traits have popped up in the D2k rules like `Buildable` on the Ornithopter, `UpdatesPlayerStatistics` on a bunch of actors, `WithSpriteBody` on the concrete, even a new actor for the death hand - what's going on with the new traits?

**Comment by Mailaender** [comment]
They are parsed by the encyclopedia logic.

**Review by PunkPun — COMMENTED** [review]
LGTM. Just one thing I noticed. IMO actors shouldn't need to have Buildable to exist in the encyclopedia, after all displaying prerequisites is optional. BuildableInfo is required only in one place so i should be easy to handle whenever BuildableInfo is missing

File: mods/d2k/rules/aircraft.yaml [hunk 29]
```diff
@@ -103,6 +109,10 @@ ornithopter:
 		HP: 9000
 	Armor:
 		Type: light
+	Encyclopedia:
+		Description: The fastest aircraft on Dune, the Ornithopther is lightly armored and capable of dropping 500lb bombs. This unit is most effective against infantry and lightly armored targets, but also damages armored targets.\n\nThey fly fast and are not targetable by anti-aircraft weapon.
```
Comment (PunkPun) [inline mods/d2k/rules/aircraft.yaml]:
```suggestion
		Description: The fastest aircraft on Dune, the Ornithopther is lightly armored and capable of dropping 500lb bombs. This unit is most effective against infantry and lightly armored targets, but also damages armored targets.
```
Ornithopters are targetable both with Missile tanks and Missile turrets

**Comment by Mailaender** [comment]
Made `Buildable` optional.

**Review by PunkPun — COMMENTED** [review]
Could you select the first element when you enter the mentat? It feels rather weird to have the page blank

**Comment by Mailaender** [comment]
Sure, good idea.

**Review by PunkPun — DISMISSED** [review]
LGTM

**Review by dragunoff — CHANGES_REQUESTED** [review]
image
Empty category looks weird - how about "Super weapons"? :thinking: I think any kind of header here would be better, even a "Misc" or "Other".

image
Caryall is listed twice.

Screenshot from 2022-08-29 20-42-43
The panel seems too small. How about making it the same size as most other panels? I know a large part of it would be empty (at least for now) but it just gives it a better feel and makes the scroll panel at the side much more usable. Feel free to pick the layout changes from here: https://github.com/dragunoff/OpenRA/commit/7aac74f8fd5b590af306c2b1f40c5c7b9f4d2b06

**Comment by Mailaender** [comment]
The layout is deliberately to avoid all the blank spaces your change would introduce.

**Comment by dragunoff** [comment]
image

> The layout is deliberately to avoid all the blank spaces your change would introduce.

The blank space is of little concern, I don't see a big downside to it. I'm against the tight layout because:
- The scroll area on the side is barely usable - it's too short and thus it's hard to see what items are available and difficult to find the item you seek
- The panel size is inconsistent with most of the other panels (I've spent a great deal of effort to unify the panel sizes - see #19681)
- It just looks crammed :shrug:

**Comment by Mailaender** [comment]
You got your enlarged layout. Even increasing the font does not solve the problem that it looks mostly empty. However, it adds space for future use. I thought about doing bar charts for the stats or something when people like this.

**Review by dragunoff — CHANGES_REQUESTED** [review]
@Mailaender Nice touch with shrinking the scrollpane :+1: I think it looks pretty good despite the empty space below and as you said it leaves room for adding stats and other stuff. Most importantly the size is consistent with the rest of the panels and the list on the left is usable.

I noticed just one small issue - the first scroll item (conyard) remains selected:
image

**Comment by Mailaender** [comment]
> I noticed just one small issue - the first scroll item (conyard) remains selected.

Fixed.

**Review by dragunoff — APPROVED** [review]

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/a3b5641ca48c35cf7e74b0c635fea75acb6947e5...803f9306ddfbf41968ad2d70894de71a85acfa36)
